tools/lint/lint.py

```diff
@@ -264,6 +264,54 @@ def check_css_globally_unique(repo_root, paths):
     return errors
 
 
+def check_owners(repo_root, paths):
+    """
+    Checks that all top-level directories have OWNERS files.
+
+    :param repo_root: the repository root
+    :param paths: list of all paths
+    :returns: a list of errors found in ``paths``
+
+    """
+    # dirs represents whether there's an OWNERS file, value is a boolean.
+    dirs = {}
+    errors = []
+    owner_in_subdirs = [
+        ".well-known",
+        "css",
+        "css/vendor-imports",
+    ]
+
+    for path in paths:
+        if os.name == "nt":
+            path = path.replace("\\", "/")
+
+        source_file = SourceFile(repo_root, path, "/")
+        parts = source_file.dir_path.split(os.path.sep)
+
+        if not parts:
+            continue  # Skip top-level files
+        level = 0
+        for dir in owner_in_subdirs:
+            if path.startswith(dir):
+                level = dir.count("/") + 1
+        if len(parts) <= level:
+            continue
+        key = "/".join(parts[:level + 1])
+        if key not in dirs:
+            dirs[key] = False
+        if path.endswith("/OWNERS") and len(parts) == level + 1:
+            dirs[key] = True
+
+    for dir in dirs:
+        if dir == "":
```

Comment:
So for the key to be `""`, `"/".join(parts[:level + 1])` needs to be `""`, which will happen if `parts[:level + 1]` is `[]` or `[""]`. 

The former can only happen when parts is [] which AFAICT is impossible (because it comes from str.split), so it must be `[""]`, which implies `source_file.dir_path` is `os.path.sep`, which it shouldn't ever be.

*digs into deeper*

Reply:
So `source_file.dir_path` is of course empty for when we have files at the top-level.

`/check_stability.ini` gives a `source_file.dir_path` of `""` and `source_file.filename` of `check_stability.ini`. So where I was wrong was saying `not parts` suffices for top-level files. I'd guess we should just go for `if not source_file.dir_path`.